modules/common/src/main/scala/com/snowplowanalytics/snowplow/postgres/shredding/Type.scala

```diff
@@ -52,61 +50,64 @@ object Type {
   case object Bool extends Type
   case object Jsonb extends Type
 
-  type DataTypeSuggestion = (Schema, String) => Option[Type]
+  type DataTypeSuggestion = Schema => Option[Type]
+
+  val DefaultVarcharSize = 4096
 
   /** Derive a Postgres type, given JSON Schema */
-  def getDataType(properties: Schema, varcharSize: Int, columnName: String, suggestions: List[DataTypeSuggestion]): Type =
+  def getDataType(properties: Schema, suggestions: List[DataTypeSuggestion]): Type =
     suggestions match {
-      case Nil => Type.Varchar(4096) // Generic
+      case Nil => Type.Varchar(DefaultVarcharSize) // Generic
       case suggestion :: tail =>
-        suggestion(properties, columnName) match {
+        suggestion(properties) match {
           case Some(format) => format
-          case None         => getDataType(properties, varcharSize, columnName, tail)
+          case None         => getDataType(properties, tail)
         }
     }
 
   // For complex enums Suggest VARCHAR with length of longest element
-  val complexEnumSuggestion: DataTypeSuggestion = (properties, _) =>
+  val complexEnumSuggestion: DataTypeSuggestion = properties =>
     properties.enum match {
       case Some(enums) if isComplexEnum(enums.value) =>
         val longest = excludeNull(enums.value).map(_.noSpaces.length).maximumOption.getOrElse(16)
         Some(Type.Varchar(longest))
       case _ => None
     }
 
-  val productSuggestion: DataTypeSuggestion = (properties, _) =>
+  val productSuggestion: DataTypeSuggestion = properties =>
     properties.`type` match {
       case Some(t: SType.Union) if t.isUnion =>
         Some(Type.Jsonb)
-      case Some(t: SType) if t === (SType.Array: SType) =>
+      case Some(SType.Array) =>
         Some(Type.Jsonb)
       case Some(SType.Union(types)) if types.contains(SType.Array) =>
         Some(Type.Jsonb)
       case _ => None
     }
 
-  val timestampSuggestion: DataTypeSuggestion = (properties, _) =>
+  val timestampSuggestion: DataTypeSuggestion = properties =>
     (properties.`type`, properties.format) match {
       case (Some(types), Some(Format.DateTimeFormat)) if types.possiblyWithNull(SType.String) =>
         Some(Type.Timestamp)
       case _ => None
     }
 
-  val dateSuggestion: DataTypeSuggestion = (properties, _) =>
+  val dateSuggestion: DataTypeSuggestion = properties =>
     (properties.`type`, properties.format) match {
       case (Some(types), Some(Format.DateFormat)) if types.possiblyWithNull(SType.String) =>
         Some(Type.Date)
       case _ => None
     }
 
-  val arraySuggestion: DataTypeSuggestion = (properties, _) =>
+  /* TODO: what is this supposed to match?  Every array type is already matched by productSuggestion */
```

Comment:
I guess it might be a copy-pasta from BigQuery Loader where `[array, null]` is a special case? Even if that's not the case - I don't see how `VARCHAR` is better than `JSONB`. Perhaps we can remove it.

Reply:
I will remove it.